### PR TITLE
Schedule term filter added and Year switching

### PR DESCRIPTION
**Pull request by AndrewPolanyi**
- Added new radio elements for term filtering
- Added colors to radio elements
- Added year drop down
- Changed COURSES query to use 'year' variable (default to 2022)
- Added message to tell user if the have not generated schedule for a given year

image

image

### CODE REVIEW

File: src/pages/Schedule.tsx [hunk 1]
```diff
@@ -103,6 +110,12 @@ export const Schedule = () => {
     return onUnmount;
   }, []);
 
+  const changeYear = (newYear: string) => {
+    if (newYear != null && newYear != "") {
```
Comment (jayden-chan) [inline src/pages/Schedule.tsx]:
```suggestion
    if (newYear !== null && newYear !== "") {
```

File: src/pages/Schedule.tsx [hunk 2]
```diff
@@ -215,123 +228,161 @@ export const Schedule = () => {
...
-              p={10}
-              paddingTop={"0.5rem"}
-              borderRadius={10}
-              flexDir="column"
-              style={{ boxShadow: "0px 0px 30px rgba(0, 0, 0, 0.40)" }}
+      <>
+        <Heading>View Schedule</Heading>
+        <FormLabel htmlFor="year">For Year:</FormLabel>
+        <Select
+          placeholder="Select Year"
```
Comment (jayden-chan) [inline src/pages/Schedule.tsx]:
```suggestion
          placeholder="Select Year"
          disabled={scheduleLoading}
```
We shouldn't allow the user to change the year while the schedule is loading

**Review by jayden-chan — APPROVED** [review]
Looks good. Personally I would have added a separate radio or selector for the terms instead of adding it to the existing radio. That way you can do combination filters like ECE courses in Spring or SENG courses in Summer. But the way you did it is alright

**Review by jayden-chan — COMMENTED** [review]
image
Also I just tested the "no schedule" case and I think this styling is a little funky. You should put the text on one line and have the button below it, all centered. I think this is worth changing before merging if we plan to demo this case

**Comment by AndrewPolanyi** [comment]
> Looks good. Personally I would have added a separate radio or selector for the terms instead of adding it to the existing radio. That way you can do combination filters like ECE courses in Spring or SENG courses in Summer. But the way you did it is alright

Yeah that makes sense, I chose to do the same radio group because i figured that they can use the search as double filtering if they want.

**Review by Adamk8 — APPROVED** [review]
Gonna approve it for now so we can do some testing. I agree that it would be nice to filter by a subject AND a term but it still works how it is

**Review by nicudy — APPROVED** [review]
Looks good!

**Pull request merged**